### ISSUE TITLE
Feature: Support dynamic secret/configmap exports

### ISSUE THREAD

**Issue by matthchr** [Azure/azure-service-operator#2555]
Sometimes the consuming application wants a value like `myredis.redis.cache.windows.net:6380`, but the problem is that we export the pieces of that string separately today, like so:

```
  operatorSpec:
    secrets:
      ...
      hostName:
        name: redis-secret
        key: hostName
      sslPort:
        name: redis-secret
        key: port
```

There's no way for the application to get the value in the format it needs. Sometimes this can be worked around by the application making a code change, but that only works if the user controls the application. Some applications (such as DAPR which the above example is from) we cannot expect the user to change. We have to support getting them the values they want in a format that makes sense.

We should consider supporting something like `formattedSecrets` (or `formattedValues`?), which would be used like so:
```
  operatorSpec:
    secrets:
      ...
      hostName:
        name: redis-secret
        key: hostName
      sslPort:
        name: redis-secret
        key: port
      formattedValues:
      - name: redis-secret
        key: hostPort
        value: "${hostName}:${sslPort}"
```

Terraform supports this more generically with [format](https://www.terraform.io/language/functions/format), which is what [ARM templates](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-string#format) does as well.

**Comment by matthchr** [comment]
Is there an existing operator we could use to do this instead of doing it ourselves.

**Comment by matthchr** [comment]
We want to do this still, but need a design

**Comment by jonasnorlund** [comment]
In general we need to have a way to get values from generated resources in a dynamic approach. Let me give one example, if we create a private endpoint we would like to get new ip address of the newly created resource so we can create a network policy that opens up communication to that ip address in the next step. 

@matthchr, this is just an example, is there any roadmap item to support fetching properties from created resources in a dynamic way, not static as it is done today? Similar to Terraform or Bicep output. This becomes important when there are resources that are dependent on each others values.

**Comment by matthchr** [comment]
This is going to be our solution to exporting things like connections strings, so closed #3929 in favor in this
